### PR TITLE
Mark field `INSTANCE` as private or deprecated for removal if method `instance()` present

### PR DESCRIPTION
**Pull request by quaff**
[Please describe here what your change is about]

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

### CODE REVIEW

**Review by gavinking — APPROVED** [review]
This looks OK to me.

File: hibernate-core/src/main/java/org/hibernate/type/descriptor/java/Immutability.java [hunk 1]
```diff
@@ -23,10 +23,10 @@ public class Immutability implements MutabilityPlan<Object> {
 	/**
 	 * Singleton access
 	 */
-	public static final Immutability INSTANCE = new Immutability();
+	private static final Immutability INSTANCE = new Immutability();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/type/descriptor/java/Immutability.java:26]: Please revert and deprecate for removal instead. This is an SPI, so we'd have to first deprecate the field.

File: hibernate-core/src/main/java/org/hibernate/type/descriptor/java/ImmutableMutabilityPlan.java [hunk 2]
```diff
@@ -21,10 +21,11 @@
  * @author Steve Ebersole
  */
 public class ImmutableMutabilityPlan<T> implements MutabilityPlan<T> {
-	public static final ImmutableMutabilityPlan INSTANCE = new ImmutableMutabilityPlan();
+	@SuppressWarnings( "rawtypes" )
+	private static final ImmutableMutabilityPlan INSTANCE = new ImmutableMutabilityPlan();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/type/descriptor/java/ImmutableMutabilityPlan.java:25]: Please revert and deprecate for removal instead. This is an SPI, so we'd have to first deprecate the field.

File: hibernate-core/src/main/java/org/hibernate/sql/results/spi/ScrollableResultsConsumer.java [hunk 3]
```diff
@@ -27,7 +27,8 @@ public class ScrollableResultsConsumer<R> implements ResultsConsumer<ScrollableR
 	/**
 	 * Singleton access to the standard scrollable-results consumer instance
 	 */
-	public static final ScrollableResultsConsumer INSTANCE = new ScrollableResultsConsumer();
+	@SuppressWarnings( "rawtypes" )
+	private static final ScrollableResultsConsumer INSTANCE = new ScrollableResultsConsumer();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/sql/results/spi/ScrollableResultsConsumer.java:31]: Please revert and deprecate for removal instead. This is an SPI, so we'd have to first deprecate the field.

**Review by beikov — CHANGES_REQUESTED** [review]
Other than what I commented, this looks good to me.

**Comment by quaff** [comment]
> Other than what I commented, this looks good to me.

done.

**Comment by beikov** [comment]
Thanks

**Pull request merged**